src/common/logger.rs

```diff
@@ -3,23 +3,74 @@ use tracing_subscriber::filter;
 use tracing_subscriber::layer::SubscriberExt;
 use tracing_subscriber::prelude::*;
 
+/// Represents the role of the logger.
+#[derive(Debug)]
+pub enum LogRole {
+    /// Same as `NodeRole::Node`.
+    Node,
+    /// Same as `NodeRole::Controller`.
+    Controller,
+    /// Same as `NodeRole::SchedulerExtender`.
+    SchedulerExtender,
+    /// Same as `NodeRole::AsyncFuse`.
+    AsyncFuse,
+    /// For testing purpose.
+    #[cfg(test)]
+    Test,
+    /// For bind mounter which is a helper command to bind mount for non-root
+    /// user.
+    #[allow(dead_code)] // /bin/bind_mounter.rs is still using this.
+    BindMounter,
+}
+
+impl From<crate::config::NodeRole> for LogRole {
+    #[inline]
+    fn from(role: crate::config::NodeRole) -> Self {
+        match role {
+            crate::config::NodeRole::Node => LogRole::Node,
+            crate::config::NodeRole::Controller => LogRole::Controller,
+            crate::config::NodeRole::SchedulerExtender => LogRole::SchedulerExtender,
+            crate::config::NodeRole::AsyncFuse => LogRole::AsyncFuse,
+        }
+    }
+}
+
+impl LogRole {
+    /// Returns the string representation of the log role.
+    #[must_use]
+    #[inline]
+    pub fn as_str(&self) -> &'static str {
+        match *self {
+            LogRole::Node => "node",
+            LogRole::Controller => "controller",
+            LogRole::SchedulerExtender => "scheduler_extender",
+            LogRole::AsyncFuse => "async_fuse",
+            #[cfg(test)]
+            LogRole::Test => "test",
+            LogRole::BindMounter => "bind_mounter",
+        }
+    }
+}
+
 /// Initialize the logger with the default settings.
 /// The log file is located at `./datenlord.log`.
 #[allow(clippy::let_underscore_must_use)]
+#[allow(clippy::needless_pass_by_value)] // Just pass a temporary value is fine.
```

Comment:
We can mark `#[derive(Clone, Copy)]` for `LogRole`.

Reply:
I think current way is fine.

Reply:
Well, it's also OK.